packages/loopback/lib/application.ts

```diff
@@ -18,20 +20,45 @@ export class Application extends Context {
       if (!ctor) {
         throw new Error(`The controller ${b.key} was not bound via .toClass()`);
       }
-
+      
       const ctorFactory = (req: http.ServerRequest, res: http.ServerResponse, operationName: string) => {
         const requestContext = new Context(this);
         requestContext.bind('controller.current.ctor').to(ctor);
         requestContext.bind('controller.current.operation').to(operationName);
         requestContext.bind('http.request').to(req);
         requestContext.bind('http.response').to(res);
-        return requestContext.get(b.key);
+        //[rashmi] TODO calling getAuthenticatedUser() here only temporary until MiddleWare design/implementation is ready
+        //the reason to call here is - this is the only place right now, where we have access to to request, app context and 
+        //also where AppController's constructor is called where we need to inject the 'user' 
+        return new Promise((resolve, reject) => {
+          const metadata = getAuthenticateMetadata(ctor.prototype, operationName);
+          var isAuthRequired;
+          if (metadata != null) {
+            isAuthRequired = true;
+          } else {
+            isAuthRequired = false;
+          }
+          this.getAuthenticatedUser(isAuthRequired, req).then((user) => {
+            this.bind('authentication.user').to(user);
+            requestContext.get(b.key).then((value) => {
+              resolve(value);
+            });
+          });
+        });
+        //end of Temp code until MiddleWare design/implementation is ready
       };
       const apiSpec = getApiSpec(ctor);
       router.controller(ctorFactory, apiSpec);
     });
   }
 
+  //[rashmi] TODO  getAuthenticatedUser function here only temporary until MiddleWare design/implementation is ready
+  public async getAuthenticatedUser(isAuthRequired: boolean, request: http.ServerRequest) : Promise<Object> {
+    const strategy = await this.get('authentication.strategy');
+    const user = await getAuthenticatedUser(isAuthRequired, request, strategy as Strategy);
```

Comment:
Please cast `strategy` to `Strategy` as soon as possible.

```
const strategy: Strategy = await this.get('authentication.strategy');
const user = await getAuthenticatedUser(isAuthRequired, request, strategy);
```

That way if we add some additional code dealing with `strategy` before calling `getAuthenticatedUser`, it will use a typed value, not `any`.